launchable/commands/helper.py

```diff
@@ -1,31 +1,39 @@
 import click
 from typing import Optional
-from ..utils.session import read_session
+from ..utils.session import read_build, read_session
 
 
-def _validate_session_and_build_name(session: Optional[str], build_name: Optional[str]):
-    if session and build_name:
-        raise click.UsageError(
-            'Only one of --build or --session should be specified')
-
-    if session is None and build_name is None:
-        raise click.UsageError(
-            'Either --build or --session has to be specified')
+def find_or_create_session(context, session: Optional[str], build_name: Optional[str], flavor=[]) -> Optional[str]:
+    """Determine the test session ID to be used.
 
+    1. If the user explicitly provides the session id via the `--session` option
+    2. If the user gives no options, the current session ID is read from the session file tied to $PWD,
+       or one is created from the current build name. See https://github.com/launchableinc/cli/pull/342
+    3. The `--build` option is legacy compatible behaviour, in which case a session gets created and tied
+       to the build.
 
-def find_or_create_session(context, session: Optional[str], build_name: Optional[str], flavor=[]) -> Optional[str]:
-    _validate_session_and_build_name(session, build_name)
+    Args:
+        session: The --session option value
+        build_name: The --build option value
+    """
     from .record.session import session as session_command
 
+    saved_build_name = read_build()
+    if build_name and saved_build_name != build_name:
+        raise click.UsageError(click.style(
+            "Given build name ({}) is different from when you ran `launchable record build --name {}`.\nMake sure to run `launchable record build --name {}` before.".format(build_name, saved_build_name, build_name), fg="yellow"))
+
     if session:
         return session
-    elif build_name:
-        session_id = read_session(build_name)
+    else:
+        if not saved_build_name:
+            raise click.UsageError(click.style(
+                "Have you run `launchable record build`?\nIf not, please do. If it was run elsewhere/earlier, please use the --session option", fg="yellow"))
```

Comment:
I modified the message and in particular not to use `build_name` as that is optional and the value might not be present. More so in the future as `--build` option is legacy.